src/Table/Table.jsx

```diff
@@ -107,6 +113,12 @@ export class Table extends Component {
       newPages.push(pageData);
     }
 
+    if (this._epoch !== startEpoch) {
+      // the table was reset while we were fetching, so we don't want to
+      // actually insert our data
+      return;
+    }
```

Comment:
@jefff and I had discussed the best way to handle outdated request promises. One more direct way to do this is to track the promise itself rather than a counter

```
async _fetchLazy(pageIdx) {
  ...
  const getDataPromise = getData(query);
  this.currentPromise = getDataPromise;
  const pageData = await getDataPromise;
  if (this.currentPromise !== getDataPromise) {
    return;
  }
  ...
}

lazyReset() {
  this.currentPromise = null;
  ...
}
```